superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -41,28 +41,26 @@ class VisualizeModal extends React.PureComponent {
     this.state = {
       chartType: CHART_TYPES[0],
       datasourceName: this.datasourceName(),
-      columns: {},
+      columns: this.setStateFromProps(),
       hints: [],
     };
   }
   componentDidMount() {
     this.validate();
   }
-  componentWillReceiveProps(nextProps) {
-    this.setStateFromProps(nextProps);
-  }
-  setStateFromProps(props) {
+  setStateFromProps() {
```

Comment:
probably need to change the function name now that it does not `setState` anymore. `getColumnFromProps`?